src/EGO Forum Enhancement.ts

```diff
@@ -880,6 +880,160 @@ function addTrashButton(before: HTMLDivElement) {
     before.parentElement?.insertBefore(trashButton, before);
 }
 
+/**
+ * Adds additional buttons to post action bars to make it easier to perform some actions.
+ */
+function addPostActionBarButtons() {
+    const threadId = window.location.href.match(/threads\/(?<post_id>\d+)/);
+    if (!threadId) return;
+
+    const posts = document.querySelectorAll(
+        ".message.message--post"
+    ) as NodeListOf<HTMLElement>;
+    if (posts.length == 0) return;
+
+    const isThreadUnapproved = document.querySelector(
+        ".blockStatus-message--moderated"
+    );
+
+    for (let i = 0; i < posts.length; i++) {
+        const post = posts[i] as HTMLElement;
+        if (!post) continue;
+
+        const actionBarSet = post.querySelector(
+            ".actionBar-set.actionBar-set--internal"
+        );
+
+        if (!actionBarSet) continue;
+
+        // There is no point in adding additional buttons on deleted posts currently.
+        const isDeleted =
+            post.classList.contains("message--deleted") ||
+            post.querySelector(".messageNotice--deleted");
+        if (isDeleted) continue;
+
+        const postId = post.dataset.content?.substring(5);
+        if (!postId) continue;
+
+        let isUnapproved = post.querySelector(".messageNotice--moderated")
+            ? true
+            : false;
+
+        let isThreadOP = false;
+
+        // Check the post counter to see if it is the original post so we can make sure we change the button action correctly.
+        if (i == 0) {
+            const attributionListElements = post.querySelectorAll(
+                ".message-attribution-opposite.message-attribution-opposite--list li"
+            );
+            for (let j = 0; j < attributionListElements.length; j++) {
+                const element = attributionListElements[j] as HTMLElement;
+
+                if (element.innerText == "#1") {
+                    isThreadOP = true;
+                    break;
+                }
+            }
+        }
+
+        if (isThreadOP && isThreadUnapproved) isUnapproved = true;
+
+        const approvalButton = document.createElement("a");
+        approvalButton.classList.add(
+            "actionBar-action",
+            "actionBar-action--menuItem"
+        );
+        approvalButton.setAttribute("tabindex", "0");
+
+        if (isThreadOP) {
+            approvalButton.innerText = isUnapproved
+                ? "Approve Thread"
+                : "Unapprove Thread";
+        } else {
+            approvalButton.innerText = isUnapproved ? "Approve" : "Unapprove";
+        }
+
+        approvalButton.onclick = () => {
+            handlePostApproval(threadId[1], postId, isUnapproved);
+        };
+
+        actionBarSet.appendChild(approvalButton);
+    }
+}
+
+/**
+ * Handles (un)approving a thread post by ID.
+ */
+async function handlePostApproval(
```

Comment:
I'd rename this to something like `setApproval` or similar, handle is generally used in this project to signify a function that handles a specific page, action, or similar. This can also be used as a utility function for other features, not just as a handling function, so it would be less confusing that way.